Initialize song configurations from song directories

tobiaslabu opened the pull request:
This PR will fix issue #84. The `songs` command will accept an option called `--init` which will make mtrack scan the given directory and create song configuration files based on the audio and MIDI files it finds.

mdwn commented:
Just wanted to let you know that I've seen this and I'll review it properly soon!

tobiaslabu commented:
I hadn't marked this PR as ready yet, because I still wanted to perform some tests with real world data. These tests have now been successful, I got some working song definitions that I could use with mtrack. 

A note on the implementation: I added `serde-yml` as a dependency to write YAML files. It is a fork of the unmaintained `serde-yaml` that we just dropped. `config-rs` only supports reading config files. 
Another approach would be to use some templating engine like `handlebars` or `upon`. Let me know if you like this idea better or if you have any other suggestions.

mdwn reviewed (approved):
This is awesome! Thanks so much for tackling this. :-) I really appreciate it!
